examples/hb-subset.example.node.js

```diff
@@ -4,12 +4,43 @@ const { readFile, writeFile } = require('fs').promises;
 const { join, extname, basename } = require('path');
 const { performance } = require('node:perf_hooks');
 
-const SUBSET_TEXT = 'abc';
-
 (async () => {
     const { instance: { exports } } = await WebAssembly.instantiate(await readFile(join(__dirname, '../hb-subset.wasm')));
-    const fileName = 'NotoSans-Regular.ttf';
-    const fontBlob = await readFile(join(__dirname, '../test/fonts/noto', fileName));
+
+    function _subset_flag(s) {
+        if (s == "--glyph-names") { return 0x00000080; }
+        if (s == "--no-layout-closure") { return 0x00000200; }
+        return 0x0;
+    }
+
+    function setSubsetFlags(input, flags) {
+        let flagValue = 0;
+        flags.forEach(function (s) {
+            flagValue |= _subset_flag(s);
+        })
+        exports.hb_subset_input_set_flags(input, flagValue);
+        // console.log('flagValue', flagValue)
+    }
+
+    // Add the keys of map to keys .
+    // https://harfbuzz.github.io/harfbuzz-hb-map.html#hb-map-keys
+    function HbMapKeys(map) {
+        const mySetPtr = exports.hb_set_create();
+        exports.hb_map_keys(map, mySetPtr);
+        const result = typedArrayFromSet(mySetPtr, Uint32Array);
+        exports.hb_set_destroy(mySetPtr);
+
+        return result;
+    }
+
+    function closureAndGetGids(input) {
+        const plan = exports.hb_subset_plan_create_or_fail(face, input);
+        const glyph_map = exports.hb_subset_plan_old_to_new_glyph_mapping(plan);
```

Comment:
You'll need to destroy the plan and map once you've extracted the glyphs so the memory for them doesn't leak.

Reply:
Good catch! I added `hb_map_destroy` and `hb_subset_plan_destroy` below.